src/common/AdaptiveStream.cpp

```diff
@@ -187,10 +187,10 @@ bool AdaptiveStream::start_stream(const uint32_t seg_offset,
       if (!pos)
         pos = 1;
     }
-    //go at least 12 secs back
     uint64_t duration(current_rep_->get_segment(pos)->startPTS_ -
                       current_rep_->get_segment(pos - 1)->startPTS_);
-    pos -= static_cast<uint32_t>((12 * current_rep_->timescale_) / duration) + 1;
+
+    pos -= static_cast<uint32_t>((tree_.live_delay_ * current_rep_->timescale_) / duration);
```

Comment:
Probably still the the `+ 1` here. Previously it has been that 12 seconds back gives us the segment which *finishes* 12 seconds back from the edge. Without the +1 we're starting the stream 1 segment forward from where it would have been before.

Reply:
oops, your right. i totally missed that!

Reply:
thinking about this, the seek code only limits to the tree_.live_delay_
(originally the 12 was hard-coded here)
https://github.com/xbmc/inputstream.adaptive/pull/761/files#diff-34d21af3c614ea3cee120df276c9c4ae95053830d7f1d3deaf009a4625409ad2L3048

so the seeking is letting us get an extra segment forward
would we not want it consistent?
the seek code only deals with the seek time as well (not segments)
so not as easy to -1 segment from that

once the cache ahead is in, live_delay_ will be larger to begin with as well which should at least get that 1 segment back

Reply:
> but i think it's important that the seeking and start playback both have the same offset
which currently they actually don't

I can't get a stream to play closer than 12 seconds from the edge. Try Kayo for example as the 4s segments divide neatly.

Looking here, we get our '-1':
https://github.com/xbmc/inputstream.adaptive/blob/bce1270c2544aaa30a33545c4e24f5228b9dd590/src/common/AdaptiveStream.cpp#L544-L545

Reply:
not sure if thats the same thing.
I think its just correcting it if the calculated segment start time is ahead of the seek time.
we would want a specific -1 for live streams only in this code somewhere.

I'm trying with latest stock IA.
Redbull TV live stream from IPTV AU.
We want to make sure where we start playback is same offset as when we ffwd.
Ignores 12s. just look at the timeline and what the maximum seconds get too.
When I start playback, it will start around 0:41 - 0:42
and the highest it goes is 0:47
Now if you mash ffwd a few times, you can get it to being playback further ahead than this
eg. starting at 0:47 and playing upto 0:52 or even further
which seems to me to be a segment ahead than initial playback
Its segments at 6s
Also, if you keep trying fwd enough on windows - you may see the bug I mentioned where the stream becomes distorted at the top. 

Now, with this version, it will start playback around 0:48 and get to around 0:52
makes sense as no longer the -1.
But when i fwd, I pretty much always get the same spot or slightly back.

Actually, I know a possible solution!
We know the segment length in start stream.
So, we could simply add 1x segment length to live_delay_
That will give you the same -1 but it will be consistant

Reply:
Ok, I'm seeing it as well. I can get to the 2nd segment from the edge with 6s segments.

Just thinking out loud here but when we start a stream we're always going to be consistent - we've just refreshed the playlist, and don't refresh again until a segment has elapsed. However when we seek, we're not necessarily performing it on the exact boundary of a segment.
Eg. stream with 10s segments. Stream is started/playlist downloaded just as a new segment has been added to the playlist. 8 seconds in we seek, playlist is refreshed and returns the same content/segments (which is now valid for another 10s), and we see the window in the timeline that's been played 'slide' to the right. So with this theory of course we can get 'closer' to the live edge...

Anyway, regarding the += seg length - we don't always have consistent segment length. HLS can vary as can some DASH configurations. 

I'm thinking maybe that changing the default from 12 is best left for Nexus/rework since we're changing the default behavior? We would be best to make it the greater of SPD and 12 seconds. In rework we can also factor in a 'preferred buffer' setting.